src/net_processing.cpp

```diff
@@ -2506,6 +2506,15 @@ void ProcessMessage(
                       pfrom.nVersion.load(), pfrom.nStartingHeight,
                       pfrom.GetId(), (fLogIPs ? strprintf(", peeraddr=%s", pfrom.addr.ToString()) : ""),
                       pfrom.m_tx_relay == nullptr ? "block-relay" : "full-relay");
+            if (pfrom.IsBlockOnlyConn()) {
+                // The node dumps:
+                // - current anchors; and
+                // - anchors that were known at startup but have not tried for connections yet
+                auto anchors_to_save = connman.GetCurrentAnchors();
```

Comment:
This logic shouldn't be in net_processing. It's all bound up in connection-layer logic rather than application-layer logic. There's nothing here that's specific to the `pnode` that we're in the context of in this `ProcessMessages()` call. I think it makes sense for this to be in `CConnman` rather than here.

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/17428#issuecomment-688252761).